### ISSUE TITLE
The state property introducted by the @State is not writable

### ISSUE THREAD

**Issue by Haixing-Hu** [jquagliatini/pinia-class#1]
The state property of Vuex is readonly and should be mutated by mutations. But the pinia state is writable directly. So the @State should introduce a writable state property.

The following is a simple example:

```javascript
import { defineStore } from 'pinia';

const useExampleStore = defineStore('example', {
  state: () => ({ value: 0}),
});

export default useExampleStore;
```

```javascript
import Vue from 'vue'
import Component from 'vue-class-component'
import { State } from 'pinia-class'
import { useExampleStore } from './store/example-store'

@Component
export default class MyComponent extends Vue {
  @State(useExampleStore)
  value;

  foo(val) {
    this.value = val;    // ERROR: the value has no setter!
  }
}
```

**Comment by jquagliatini** [comment]
Hi @Haixing-Hu, thanks for your feedback. To be honest, I don't maintain actively this lib. It was mainly intended for my team to progressively migrate to pinia. Now that vue3 is more widely used, I would consider deprecating class components. 

However, if you feel like it, I wouldn't mind your help with a pull request if you need the update. There is no real complexity in this lib per se, I took inspiration from vuex-class.

**Comment by jquagliatini** [comment]
Hi @Haixing-Hu, I took some time to think about your proposal.

For context, pinia has a [mapWritableState](https://pinia.vuejs.org/core-concepts/state.html#modifiable-state) which exposes the writable state. However, it's not possible to expose getters (which is logical). 

At the moment I see 2 solutions:

1. We could replace the `@State` implementation to expose the writable state
2. Or introduce a `@WritableState` decorator

The first solution is a bit more idiomatic for vuex-class users, but not for pinia users. On the other hand, the second solution introduces a new concept, not really visible when using pinia with the composition API. 

I prefer the new decorator, but I would be curious to know your opinion on the matter. 

To be more concrete I opened a pull request with a proposal: https://github.com/jquagliatini/pinia-class/pull/3
